aggregator/src/test/java/org/jenkinsci/plugins/workflow/BuildQueueTasksTest.java

```diff
@@ -0,0 +1,192 @@
+                scheduleAndWaitQueued(p);
+                // Ok, the item is in he queue now, restart
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                WorkflowRun b = p.getBuildByNumber(1);
+
+                assertQueueAPIStatusOKAndAbort(b);
+            }
+        });
+    }
+
+    @Issue("JENKINS-28649")
+    @Test public void computerAPI() throws Exception {
+        // This is implicitly testing ExecutorStepExecution$PlaceholderTask$PlaceholderExecutable as exported bean
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                        "  echo 'test'\n " +
+                        "  semaphore 'watch'\n " +
+                        "}"));
+
+                WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+                SemaphoreStep.waitForStart("watch/1", b);
+
+                assertComputerAPIStatusOK();
+
+                SemaphoreStep.success("watch/1", null);
+                story.j.waitUntilNoActivity();
+            }
+        });
+    }
+
+    @Issue("JENKINS-28649")
+    @Test public void computerAPIRestartable() throws Exception {
+        // This is implicitly testing AfterRestartTask#run as exported bean
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                        "  echo 'test'\n " +
+                        "  semaphore 'watch'\n " +
+                        "}"));
+
+                WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+                SemaphoreStep.waitForStart("watch/1", b);
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                assertComputerAPIStatusOK();
+
+                SemaphoreStep.success("watch/1", null);
+                story.j.waitUntilNoActivity();
+            }
+        });
+    }
+
+    private WorkflowRun scheduleAndWaitQueued(WorkflowJob p) throws InterruptedException, ExecutionException {
+        QueueTaskFuture<WorkflowRun> build = p.scheduleBuild2(0);
+
+        WorkflowRun b = build.getStartCondition().get();
+        int secondsWaiting = 0;
+        while (true) {
+            if (secondsWaiting > 5) {
+                assertTrue("No item queued after 5 seconds", false);
+            }
+            if (story.j.jenkins.getQueue().getItems().length > 0) {
```

Comment:
Easier like it is - given the conditions - as `p` is not  what I want to check in the queue but the `PlaceholderTask` instance and I don't see an straightforward way to retrieve it.